### PR TITLE
Replace gun slot icons with X or tool slot icons.

### PR DESCRIPTION
**Pull request by dperelman**
This is an initial version of #51. It replaces the gun/ammo slot icons with X's (actually the `set-bar-slot` icon). It also replaces the text for the tooltips and errors when you try to equip a non gun/ammo in those slots with empty strings, effectively disabling those tooltips and errors.

If you want it to indicate that those are "tool" slots for mods that put tools in the gun slots, maybe you could use the old pickaxe icon I pulled out of an old Factorio's graphics directory:
slot-icon-tool
Not sure what the appropriate tool "ammo" icon would be.

Possibly the choice of which icon could be configurable? EDIT: Actually, you could probably just check if there exist any guns/ammo that haven't been removed and chose the icon based on that. EDIT2: Done.

### CODE REVIEW

**Comment by dperelman** [comment]
I tried to see if there was a way to programmatically change the strings based on whether there exists tool "guns"/"ammo" or not, but I don't see a way to do so. (At least some localized strings can be changed, but I don't see options for those ones.)

File: locale/en/Pacifist.cfg [hunk 1]
```diff
@@ -1,3 +1,18 @@
+[gui]
+guns=
+weapons=
+ammo=
+
+[inventory-full-message]
+guns=
```
Comment (Derpumu) [inline locale/en/Pacifist.cfg:7]: Same as above, I'd rather have a different message than none. 
How about "Tool" and "Tool ammo" or "Tool charges"? See also tooltips for ammo:
image

File: locale/en/Pacifist.cfg [hunk 2]
```diff
@@ -1,3 +1,18 @@
...
+ammo=
+
+[inventory-restriction]
+cant-be-used-as-ammo-for=
```
Comment (Derpumu) [inline locale/en/Pacifist.cfg:11]: I could not reproduce these messages, where are they used?

File: functions.lua [hunk 3]
```diff
@@ -229,6 +229,21 @@ function PacifistMod.remove_military_entities()
     end
 end
 
+function PacifistMod.disable_gun_slots()
+    local x_icon = "__core__/graphics/set-bar-slot.png"
+    local tool_icon = "__Pacifist__/graphics/slot-icon-tool.png"
+    local icon_kinds = { "gun", "ammo" }
+    for _, kind in pairs(icon_kinds) do
+        local icon = x_icon
```
Comment (Derpumu) [inline functions.lua:237]: I love the idea. I am undecided whether we should just keep it simple and always use the tool icon

**Review by Derpumu — CHANGES_REQUESTED** [review]

File: locale/en/Pacifist.cfg (hunk 2 above)
Comment (dperelman) [reply locale/en/Pacifist.cfg:11]: If you have an item on your cursor and click on the slot, then those show as floating text. I'll add them back in reworded to refer to tools.

File: locale/en/Pacifist.cfg (hunk 1 above)
Comment (dperelman) [reply locale/en/Pacifist.cfg]: I had missed that one. Now that "Ammo" says "Charges".

File: locale/en/Pacifist.cfg [hunk 4]
```diff
@@ -1,3 +1,18 @@
+[gui]
+guns=
```
[the remark this answers is not in the document]
Comment (dperelman) [reply locale/en/Pacifist.cfg]: Changed to "Tool" and "Tool charges".

File: functions.lua (hunk 3 above)
Comment (dperelman) [reply functions.lua:237]: I thought it would be confusing to have a slot that can never be used without an indication that it's disabled somehow.

**Review by Derpumu — APPROVED** [review]

**Pull request merged**